"use strict";(self.webpackChunk_N_E=self.webpackChunk_N_E||[]).push([[3594],{13594:(e,t,o)=>{o.d(t,{resendSignUpCode:()=>A});var r,i=o(42656),n=o(62468),a=o(96778),s=o(99643),l=o(69895),d=o(6929),u=o(23251),g=o(11719),c=o(83507),h=o(49240),f=o(87395),p=o(83578),m=o(74267);let y=e=>(0,c.A)(h.e,(0,f._)("ResendConfirmationCode"),(0,p.T)(),{...m.C,...e});var w=o(40850);async function A(e){let{username:t}=e;(0,s.M)(!!t,l.P.EmptySignUpUsername);let o=i.H.getConfig().Auth?.Cognito;(0,n.$y)(o);let{userPoolClientId:r,userPoolId:c,userPoolEndpoint:h}=o,f=e.options?.clientMetadata,p=(0,g.d)({username:t,userPoolId:c,userPoolClientId:r}),m=y({endpointResolver:(0,w.o)({endpointOverride:h})}),{CodeDeliveryDetails:A}=await m({region:(0,d.J)(o.userPoolId),userAgentValue:(0,u.A)(a.BM.ResendSignUpCode)},{Username:t,ClientMetadata:f,ClientId:o.userPoolClientId,UserContextData:p}),{DeliveryMedium:E,AttributeName:C,Destination:I}={...A};return{destination:I,deliveryMedium:E,attributeName:C||void 0}}var E=o(40323),C=o(58192),I=o(39421),S=o(30017),v=o(11802),_=o(43425),U=o(84986),R=o(76545),O=o(14219),T=o(11826),M=o(87410),k=o(73140);new M.C("AuthError");let b={oauthSignInError:{message:k.fs.OAUTH_ERROR,log:"Make sure Cognito Hosted UI has been configured correctly"},noConfig:{message:k.fs.DEFAULT_MSG,log:`
            Error: Amplify has not been configured correctly.
            This error is typically caused by one of the following scenarios:

            1. Make sure you're passing the awsconfig object to Amplify.configure() in your app's entry point
                See https://aws-amplify.github.io/docs/js/authentication#configure-your-app for more information
            
            2. There might be multiple conflicting versions of amplify packages in your node_modules.
				Refer to our docs site for help upgrading Amplify packages (https://docs.amplify.aws/lib/troubleshooting/upgrading/q/platform/js)
        `},missingAuthConfig:{message:k.fs.DEFAULT_MSG,log:`
            Error: Amplify has not been configured correctly. 
            The configuration object is missing required auth properties.
            This error is typically caused by one of the following scenarios:

            1. Did you run \`amplify push\` after adding auth via \`amplify add auth\`?
                See https://aws-amplify.github.io/docs/js/authentication#amplify-project-setup for more information

            2. This could also be caused by multiple conflicting versions of amplify packages, see (https://docs.amplify.aws/lib/troubleshooting/upgrading/q/platform/js) for help upgrading Amplify packages.
        `},emptyUsername:{message:k.fs.EMPTY_USERNAME},invalidUsername:{message:k.fs.INVALID_USERNAME},emptyPassword:{message:k.fs.EMPTY_PASSWORD},emptyCode:{message:k.fs.EMPTY_CODE},signUpError:{message:k.fs.SIGN_UP_ERROR,log:"The first parameter should either be non-null string or object"},noMFA:{message:k.fs.NO_MFA},invalidMFA:{message:k.fs.INVALID_MFA},emptyChallengeResponse:{message:k.fs.EMPTY_CHALLENGE},noUserSession:{message:k.fs.NO_USER_SESSION},deviceConfig:{message:k.fs.DEVICE_CONFIG},networkError:{message:k.fs.NETWORK_ERROR},autoSignInError:{message:k.fs.AUTOSIGNIN_ERROR},default:{message:k.fs.DEFAULT_MSG}};var N=o(35346);let P=(e,t)=>new N.l({message:e??"An error has occurred during the oauth process.",name:k.nC.OAuthSignInError,recoverySuggestion:t??b.oauthSignInError.log});!function(e){e.NoConfig="noConfig",e.MissingAuthConfig="missingAuthConfig",e.EmptyUsername="emptyUsername",e.InvalidUsername="invalidUsername",e.EmptyPassword="emptyPassword",e.EmptyCode="emptyCode",e.SignUpError="signUpError",e.NoMFA="noMFA",e.InvalidMFA="invalidMFA",e.EmptyChallengeResponse="emptyChallengeResponse",e.NoUserSession="noUserSession",e.Default="default",e.DeviceConfig="deviceConfig",e.NetworkError="networkError",e.AutoSignInError="autoSignInError",e.OAuthSignInError="oauthSignInError"}(r||(r={}));let j=async e=>{let t=await I.f.loadOAuthState(),o=e===t?t:void 0;if(!o)throw new N.l({name:r.OAuthSignInError,message:null===e?"`signInWithRedirect` has been canceled.":"An error occurred while validating the state.",recoverySuggestion:null===e?void 0:"Try to initiate an OAuth flow from Amplify"});return o},D=async({currentUrl:e,userAgentValue:t,clientId:o,redirectUri:r,responseType:i,domain:n,preferPrivateSession:a})=>{let s=new S.o(e),l=s.searchParams.get("error"),d=s.searchParams.get("error_description");if(l)throw P(d??l);return"code"===i?F({currentUrl:e,userAgentValue:t,clientId:o,redirectUri:r,domain:n,preferPrivateSession:a}):G({currentUrl:e,redirectUri:r,preferPrivateSession:a})},F=async({currentUrl:e,userAgentValue:t,clientId:o,redirectUri:r,domain:i,preferPrivateSession:a})=>{let s=new S.o(e),l=s.searchParams.get("code"),d=s.searchParams.get("state");if(!l||!d)throw P("User cancelled OAuth flow.");let u=await j(d),g=await I.f.loadPKCE(),c=Object.entries({grant_type:"authorization_code",code:l,client_id:o,redirect_uri:r,...g?{code_verifier:g}:{}}).map(([e,t])=>`${encodeURIComponent(e)}=${encodeURIComponent(t)}`).join("&"),{access_token:h,refresh_token:f,id_token:p,error:m,error_message:y,token_type:w,expires_in:A}=await (await fetch("https://"+i+"/oauth2/token",{method:"POST",headers:{"Content-Type":"application/x-www-form-urlencoded",[v.YJ]:t},body:c})).json();if(m)throw P(y??m);let E=(h&&(0,n.Cq)(h).payload.username)??"username";return await (0,U.i)({username:E,AccessToken:h,IdToken:p,RefreshToken:f,TokenType:w,ExpiresIn:A}),Y({redirectUri:r,state:u,preferPrivateSession:a})},G=async({currentUrl:e,redirectUri:t,preferPrivateSession:o})=>{let{id_token:r,access_token:i,state:a,token_type:s,expires_in:l,error_description:d,error:u}=(new S.o(e).hash??"#").substring(1).split("&").map(e=>e.split("=")).reduce((e,[t,o])=>({...e,[t]:o}),{id_token:void 0,access_token:void 0,state:void 0,token_type:void 0,expires_in:void 0,error_description:void 0,error:void 0});if(u)throw P(d??u);if(!i)throw P("No access token returned from OAuth flow.");let g=await j(a),c=(i&&(0,n.Cq)(i).payload.username)??"username";return await (0,U.i)({username:c,AccessToken:i,IdToken:r,TokenType:s,ExpiresIn:l}),Y({redirectUri:t,state:g,preferPrivateSession:o})},Y=async({redirectUri:e,state:t,preferPrivateSession:o})=>{await T.o.setOAuthMetadata({oauthSignIn:!0}),await I.f.clearOAuthData(),await I.f.storeOAuthSignIn(!0,o),(0,O.i)(),L(t)&&_.YZ.dispatch("auth",{event:"customOAuthState",data:(W(t).match(/.{2}/g)||[]).map(e=>String.fromCharCode(parseInt(e,16))).join("")},"Auth",_.U9),_.YZ.dispatch("auth",{event:"signInWithRedirect"},"Auth",_.U9),await (0,R.Q)(),q(e)},L=e=>/-/.test(e),W=e=>e.split("-").splice(1).join("-"),q=e=>{"undefined"!=typeof window&&void 0!==window.history&&window.history.replaceState(window.history.state,"",e)};var H=o(65410);let V=async e=>{(0,O.i)(),await I.f.clearOAuthInflightData(),_.YZ.dispatch("auth",{event:"signInWithRedirect_failure",data:{error:e}},"Auth",_.U9)},$=async e=>{try{(0,n.$y)(e),(0,n.G6)(e),I.f.setAuthConfig(e)}catch(e){return}if(await I.f.loadOAuthInFlight())try{let t=window.location.href,{loginWith:o,userPoolClientId:r}=e,{domain:i,redirectSignIn:n,responseType:s}=o.oauth,l=(0,H.M)(n);await D({currentUrl:t,clientId:r,domain:i,redirectUri:l,responseType:s,userAgentValue:(0,u.A)(a.BM.SignInWithRedirect)})}catch(e){await V(e)}};(0,E.B)()&&i.H[C.$]($);o(49570)}}]);